### DISCUSSION TITLE
[HPM] Error occurred while trying to proxy request

### DISCUSSION THREAD

**Discussion by minsforgh** [devrel-kr/HackaLearn#60] [2 Q&A - 클라우드 스킬 챌린지] [answered]
클라우드 챌린지 " Azure Static Web Apps를 통한 사용자 인증 / 연습 - 사용자 인증 추가  " 수행 중 문제 발생하여 도움 요청합니다.
"로컬에서 앱 실행하기"를 수행 중 터미널에서 "npm start" 를 입력 시  " [HPM] Error occurred while trying to proxy request /api/products/ from localhost:4200 to http://localhost:7071 (ECONNREFUSED) " 라는 에러 메세지가 표시됩니다. 

![image](https://user-images.githubusercontent.com/83274084/128377362-06f46918-5895-46a6-965f-5fbf88a51f57.png)

무시하고 진행 하였을 시에는
![image](https://user-images.githubusercontent.com/83274084/128377065-9e1614c5-aa84-420c-9141-eb0ea4ca28d3.png)
다음과 같이 웹앱에서 로그인할 ID 공급자 중 하나를 선택하여도, 페이지가 제대로 리디렉션되지 않는 문제가 발생합니다.
앞서 언급한 에러가 이것의 원인인지, 그 해결방안은 무엇인지 알고 싶습니다.

**Comment by justinyoo** [comment, marked as answer]
애저 펑션 API를 실행시키지 않아서 생기는 문제입니다. 해결 방법은 아래와 같습니다.

1. 터미널 창을 하나 연다.
2. `cd api` 명령어로 API 디렉토리로 이동한다.
3. `func start` 명령어로 애저 펑션 API를 실행한다.
4. 새 터미널 창을 연다.
5. `cd angular-app` 명령어로 Angular 앱으로 이동한다.
6. `npm start` 명령어로 앱을 실행한다.

이렇게 하면 문제가 해결될 거예요.
